### PR TITLE
add option to explicitly set api base url query result location

### PR DESCRIPTION
**Pull request by juliangruendner**

### CODE REVIEW

File: src/main/java/de/numcodex/feasibility_gui_backend/api/QueryHandlerRestController.java [hunk 1]
```diff
@@ -54,6 +58,16 @@ public Response runQuery(
             .pathSegment("api", "v1", "query-handler", "result", id)
             .build()
             .toUri();
+
+    if(apiBaseUrl.length() > 0){
```
Comment (DiCanio) [inline src/main/java/de/numcodex/feasibility_gui_backend/api/QueryHandlerRestController.java:62]: The value parameter within the constructor does not have a default value. Thus, this will throw a `NullPointerException` if `app.apiBaseUrl` is simply not defined.

File: src/main/java/de/numcodex/feasibility_gui_backend/api/QueryHandlerRestController.java [hunk 2]
```diff
@@ -31,9 +32,12 @@
 public class QueryHandlerRestController {
 
   private final QueryHandlerService queryHandlerService;
+  private final String apiBaseUrl;
 
-  public QueryHandlerRestController(QueryHandlerService queryHandlerService) {
+  public QueryHandlerRestController(QueryHandlerService queryHandlerService,
+      @Value("${app.apiBaseUrl}") String apiBaseUrl) {
     this.queryHandlerService = queryHandlerService;
+    this.apiBaseUrl = apiBaseUrl;
```
Comment (DiCanio) [inline src/main/java/de/numcodex/feasibility_gui_backend/api/QueryHandlerRestController.java:40]: `app.apiBaseUrl` does not have a default value and thus can be null. This should be avoided where possible.
So, we should either define an appropriate default value matching the implementation or at least introduce a non null check.

A check could look like this:
```
this.apiBaseUrl = Objects.requireNonNull(apiBaseUrl)
```

File: src/main/java/de/numcodex/feasibility_gui_backend/api/QueryHandlerRestController.java [hunk 3]
```diff
@@ -54,6 +58,16 @@ public Response runQuery(
...
+
+    if(apiBaseUrl.length() > 0){
+      uri =
+          ServletUriComponentsBuilder.fromUriString(apiBaseUrl)
+              .replacePath("")
+              .pathSegment("api", "v1", "query-handler", "result", id)
+              .build()
+              .toUri();
+    }
+
```
Comment (DiCanio) [inline src/main/java/de/numcodex/feasibility_gui_backend/api/QueryHandlerRestController.java:70]: Reassigning another value to `uri` is unnecessary.
Instead we can shorten this to something like:

```suggestion
UriComponentsBuilder uriBuilder = (apiBaseUrl != null && !apiBaseUrl.isEmpty())
            ? ServletUriComponentsBuilder.fromUriString(apiBaseUrl)
            : ServletUriComponentsBuilder.fromRequestUri(httpServletRequest);

    var uri = uriBuilder.replacePath("")
            .pathSegment("api", "v1", "query-handler", "result", id)
            .build()
            .toUri();
```

File: src/main/resources/application.yml [hunk 4]
```diff
@@ -44,6 +44,7 @@ app:
   conceptTreeFile: ${CONCEPT_TREE_FILE:ontology/conceptTree.json}
   fhirTranslationEnabled: ${FHIR_TRANSLATE_ENABLED:false}
   cqlTranslationEnabled: ${CQL_TRANSLATE_ENABLED:true}
+  apiBaseUrl: ${API_BASE_URL:}
```
Comment (DiCanio) [inline src/main/resources/application.yml:47]: Please also add this to the `README.md` file alongside an explanation on what this does.

File: src/main/java/de/numcodex/feasibility_gui_backend/api/QueryHandlerRestController.java (hunk 2 above)
Comment (DiCanio) [reply src/main/java/de/numcodex/feasibility_gui_backend/api/QueryHandlerRestController.java:40]: Just saw the default value within the `application.yml`. A check would not hurt nonetheless 😃

**Review by DiCanio — CHANGES_REQUESTED** [review]
Adding to the review comments:
It would be nice if the implemented behavior was checked within a short unit test.

File: src/main/java/de/numcodex/feasibility_gui_backend/api/QueryHandlerRestController.java (hunk 1 above)
Comment (juliangruendner) [reply src/main/java/de/numcodex/feasibility_gui_backend/api/QueryHandlerRestController.java]: fixex with your suggested change below
